cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java

```diff
@@ -219,26 +219,29 @@ private static CpuCapacity getCpuBasedOnRequirements(ResourceRequirements resour
         Integer limit = getResourceRequirement(resourceRequirements, ResourceRequirementType.LIMIT);
 
         if (limit != null) {
-            if (request == null || limit.intValue() == request.intValue()) {
+            if (request != null) {
+                return new CpuCapacity(CpuCapacity.milliCpuToCpu(request));
+            } else {
                 return new CpuCapacity(CpuCapacity.milliCpuToCpu(limit));
             }
+        } else if (request != null) {
+            return new CpuCapacity(CpuCapacity.milliCpuToCpu(request));
         }
```

Comment:
I think I raised this already in the last review -> why not use something like this?

```
if (request != null) {
    // use request
} else if (limit != null) {
    // Use limit
} else {
    // use null
}
```

Reply:
Now refactored as suggested